### PR TITLE
Fixed the bug that prevented shapefiles from being displayed 

### PR DESCRIPTION
**Pull request by SteveMHoward**
Fixed a bug that was preventing shapefiles from being displayed on the front end. Also fixed a minor issue where they weren't being displayed on local dev environments (looking for localhost:4300 rather than localhost:3000).

### CODE REVIEW

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 1]
```diff
@@ -166,6 +173,7 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
       }.bind(this), 500);
       shapefile.addTo(this.map);
     } else if (this.project) {
+      console.log('No shapefille!', this.shapefile[0]);
```
Comment (CR-gbc) [inline src/app/project/project-details-tab/project-details-tab.component.ts]: is this a debugging item left in?

not sure on this project's policy of logging to console.

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 2]
```diff
@@ -44,13 +47,17 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
     this.commentPeriod = this.project.commentPeriodForBanner;
     this.route.data.subscribe((res: any) => {
       if (res) {
-        if (res.shapefile && res.shapefile[0].data.meta && res.shapefile[0].data.meta.length > 0) {
-          this.shapefile = res.shapefile[0].data.searchResults;
+        if (res.documents && res.documents[0].data.meta && res.documents[0].data.meta.length > 0) {
+          this.shapefile = res.documents[0].data.searchResults;
```
Comment (ShawnTurple) [inline src/app/project/project-details-tab/project-details-tab.component.ts:50]: You might want to consider using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining here, might clean up the code a bit

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 3]
```diff
@@ -44,13 +47,17 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
...
         } else {
           this.shapefile = [];
         }
       }
     });
+    // The following items are loaded by a file that is only present on cluster builds.
+    // Locally, this will be empty and local defaults will be used.
+    const remote_api_path = window.localStorage.getItem('from_public_server--remote_api_base_path');
```
Comment (ShawnTurple) [inline src/app/project/project-details-tab/project-details-tab.component.ts]: This is the WordPress php in you coming out try `const remoteApiPath =`

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 4]
```diff
@@ -156,7 +163,7 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
     if (this.shapefile[0] && this.shapefile[0].documentFileName && this.shapefile[0].documentFileName.length > 0) {
```
Comment (ShawnTurple) [inline src/app/project/project-details-tab/project-details-tab.component.ts]: again might want to consider The optional chaining operator (?.)

**Review by CR-gbc — COMMENTED** [review]
i am not the console.log police, so other than the log, all seems well. I'll leave it up to you if you want to leave it in.

File: src/app/project/project-details-tab/project-details-tab.component.ts [hunk 5]
```diff
@@ -156,7 +163,7 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
...
       console.log('Shapefile', this.shapefile);
       const escapedName = encode(this.shapefile[0].documentFileName).replace(/\(/g, '%28').replace(/\)/g, '%29').replace(/\\/g, '_').replace(/\//g, '_').replace(/\%2F/g, '_');
-      const shapeurl = '/api/document/' + this.shapefile[0]._id + '/fetch/' + escapedName;
+      const shapeurl = this.pathAPI + '/document/' + this.shapefile[0]._id + '/fetch/' + escapedName;
```
Comment (ShawnTurple) [inline src/app/project/project-details-tab/project-details-tab.component.ts:165]: are you sure there will always be a shapefile._id here...

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 1 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts]: Good call! I removed others but missed that one!

**Comment by CR-gbc** [comment]
you should have seen my eyes light up when i saw a @SteveMHoward commit!

File: src/app/project/project-details-tab/project-shapefile-resolver.service.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ export class ShapeFileResolver implements Resolve<Observable<object>> {
       currentPage,
       pageSize,
       sortBy,
-      {},
+      { documentSource: 'SHAPEFILE' }, // only look for shapefiles so we are not needlessly parsing through images or other documents
```
Comment (ShawnTurple) [inline src/app/project/project-details-tab/project-shapefile-resolver.service.ts]: Comments should start with caps and end with full stops (ie .?!)

**Review by ShawnTurple — CHANGES_REQUESTED** [review]
As far as using the the optional chaining operator (?.) .. your call it would make code a bit more readable.
NW:
* Fix comment begins with cap, ends full stop.
* Naming conventions should be camelCase

**Comment by SteveMHoward** [comment]
@ShawnTurple Good calls! Still got to shake of that rust a bit ;)

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 3 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts]: Fixed in latest push

**Review by ShawnTurple — APPROVED** [review]

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 2 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts:50]: Considered, but not willing to change at this time. Could be code impacts for the empty array that is set below vs whatever happens to the variable when const something?.something else fails.

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 1 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts]: Fixed in latest push.

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 4 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts]: Not implementing for now. Would have to redo the entire conditional function based on this change, which I feel is out of scope for a bug fix.

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 5 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-details-tab.component.ts:165]: Added a check for _id :)

File: src/app/project/project-details-tab/project-shapefile-resolver.service.ts (hunk 6 above)
Comment (SteveMHoward) [reply src/app/project/project-details-tab/project-shapefile-resolver.service.ts]: Fixed!

File: src/app/project/project-details-tab/project-details-tab.component.ts (hunk 2 above)
Comment (Baelx) [reply src/app/project/project-details-tab/project-details-tab.component.ts:50]: @SteveMHoward Actually, I don't think there's a downside to using optional chaining in this instance. The if statements will still evaluate the same. The difference is:

- Without optional chaining: Any one of those properties missing will cause a nasty error to be thrown saying that there is no such property. We just haven't run into this before. The if statement will _not_ default to the "else" but the code will error out.
- With optional chaining: If any property is missing, "undefined" will be output and the if statement logic will continue as normal.

So, TLDR, optional chaining will actually make it _more_ likely the if statement logic evaluates as expected.
